File: src/main/java/no/nav/veilarbaktivitet/aktivitet/domain/EksternAktivitetData.java
package no.nav.veilarbaktivitet.aktivitet.domain;

import lombok.Builder;
import lombok.Data;
import lombok.Singular;
import lombok.With;
import no.nav.veilarbaktivitet.aktivitetskort.*;
import no.nav.veilarbaktivitet.aktivitetskort.dto.Attributt;
import no.nav.veilarbaktivitet.aktivitetskort.dto.Etikett;
import no.nav.veilarbaktivitet.aktivitetskort.dto.LenkeSeksjon;
import no.nav.veilarbaktivitet.aktivitetskort.dto.Oppgaver;
import no.nav.veilarbaktivitet.arena.model.ArenaId;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder(toBuilder = true)
@With
public class EksternAktivitetData {
    String source;
    String tiltaksKode;
    boolean opprettetSomHistorisk;
    LocalDateTime oppfolgingsperiodeSlutt;
    ArenaId arenaId;
    AktivitetskortType type;
    Oppgaver oppgave;
    @Singular("handling")
    List<LenkeSeksjon> handlinger;
    @Singular("detalj")
    List<Attributt> detaljer;
    @Singular("etikett")
    List<Etikett> etiketter;
}

File: src/main/kotlin/no/nav/veilarbaktivitet/aktivitet/domain/EksternAktivitetData.kt
package no.nav.veilarbaktivitet.aktivitet.domain

import lombok.Singular
import no.nav.veilarbaktivitet.aktivitetskort.dto.AktivitetskortType
import no.nav.veilarbaktivitet.aktivitetskort.dto.aktivitetskort.Attributt
import no.nav.veilarbaktivitet.aktivitetskort.dto.aktivitetskort.Etikett
import no.nav.veilarbaktivitet.aktivitetskort.dto.aktivitetskort.LenkeSeksjon
import no.nav.veilarbaktivitet.aktivitetskort.dto.aktivitetskort.Oppgaver
import no.nav.veilarbaktivitet.arena.model.ArenaId
import java.time.LocalDateTime
import java.time.ZonedDateTime

data class EksternAktivitetData(
    val source: String? = null,
    val tiltaksKode: String? = null,
    val opprettetSomHistorisk: Boolean = false,
    val oppfolgingsperiodeSlutt: LocalDateTime? = null,
    val arenaId: ArenaId? = null,
    val type: AktivitetskortType,
    val oppgave: Oppgaver? = null,
    @Singular("handling")
    val handlinger: List<LenkeSeksjon>? = null,
    @Singular("detalj")
    val detaljer: List<Attributt>? = null,
    @Singular("etikett")
    val etiketter: List<Etikett>? = null,
    val endretTidspunktKilde: ZonedDateTime? = null
) {}
